### PR TITLE
Allow commands from pr description

### LINKED ISSUE #239: Allow `merge` commands in PR text
Currently, we have to comment on a PR for it to be merged by hoff. It would be nice to be able to issue merge commands directly when creating the PR, in the case of automated changes that can be merged without reviewing. We already receive and parse the pull request opened event. 

https://github.com/channable/hoff/blob/94d90e7ae919429f4a92536283f7638159d899cb/src/Github.hs#L70-L80

To support this, we just need to additionally parse the [`body` field of `pull_request`](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=opened#pull_request) and process any merge commands found. 
Function that gets called when a pull request is handled can be found [here](https://github.com/channable/hoff/blob/94d90e7ae919429f4a92536283f7638159d899cb/src/Logic.hs#L397-L407) and the command parsing function is [here](https://github.com/channable/hoff/blob/94d90e7ae919429f4a92536283f7638159d899cb/src/Logic.hs#L569-L587).

### PR DESCRIPTION
**Pull request by musicismyalibi**
Allow merge commands to be part of the pull request description by handling the description of the pull request as an added comment.

resolves #239

### CODE REVIEW

File: src/Github.hs
Comment (crtschin) [inline src/Github.hs:80]: Can you add a newtype for Text, it conveyes a bit more meaning. Might also be best to stick to Github's API spec, with the `body` being something like `Maybe Body`.

File: src/Github.hs
Comment (crtschin) [inline src/Github.hs:156]: According to Github's [own docs](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=opened#pull_request), the `pull_request.body` key will always be present. It will just be `null` if the PR description was empty.

This combined with the change of `body` being `Maybe Text`, and I believe you can use `getNested` like the other cases.

File: src/Logic.hs
Comment (crtschin) [inline src/Logic.hs:410]: I don't think you need to create a whole new function for this. This would be the only usesite of `handlePullRequestOpened`. 

Instead you can insert the pull request into the project's state and handle whatever was in the description, if there was anything, in a single function.

**Review by crtschin — CHANGES_REQUESTED** [review]
Logic seems alright. 

This has a funny side-effect though. I believe if a user would close a PR (that had a merge command in its description) before it was merged and reopen it again, it would still initiate the same merge command. This is probably fine, but it would be good to make this behavior explicit in a test.

Can you also add tests where you start out from an empty project state and insert a pull request in using the events. That way we can also test what the bot will do.

File: src/Logic.hs
Comment (musicismyalibi) [reply src/Logic.hs:410]: I created this separate function because it is not the only usesite of `handlePullRequestOpened`. It is also used in `clearPullRequest`: 
https://github.com/channable/hoff/blob/72511a0afe696f85d64003164de926d7c33a0f1b/src/Logic.hs#L362-L373

Comment (crtschin) [reply src/Logic.hs:410]: Ah missed that. My mistake

**Comment by fatho** [comment]
> I believe if a user would close a PR (that had a merge command in its description) before it was merged and reopen it again, it would still initiate the same merge command. This is probably fine, but it would be good to make this behavior explicit in a test.

I think it would be better to avoid that, since at that point someone is already manually interacting with the PR anyway, and this indeed sounds quite surprising. There's probably a way of distinguishing PR creations from PR re-openings, right?

File: src/EventLoop.hs
Comment (crtschin) [inline src/EventLoop.hs:60]: While this does solve hoff from parsing merge commands from PR descriptions when it is closed and reopened. This way of doing it is a bit hidden and implicit. 

Can you add a descriptive comment on why we remove the PR body from the event, how this affects the parsing of merge commands in descriptions, and why we want to avoid that behavior on PR reopens.

File: src/EventLoop.hs
Comment (crtschin) [inline src/EventLoop.hs:63]:
```suggestion
      -- If the pull request is reopened, we do not want to parse the command from
      -- the body, because it means that a person manually intervened with the pull 
      -- request. This is why we do not pass the body to the even, so we don't 
      -- accidentally repeatedly process the same command.
```

**Review by crtschin — COMMENTED** [review]
One small nit, can you also add a line to the changelog (and version)?

File: src/EventLoop.hs
Comment (crtschin) [inline src/EventLoop.hs:62]: Oopsie, had a typo in my suggestion: `s/even/event`

**Review by crtschin — APPROVED** [review]
One last minuscule nit, and then :shipit:

**Comment by musicismyalibi** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @musicismyalibi, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as 75c74664cb3c92c452bffd52b412ab40da648adc, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/48ff59b6-a3ff-4749-bfe6-4173b0d1ea59?pipeline_id=951626dd-296b-4ff8-ba1e-2ee72c25e146) started.

**Pull request merged**
